EIPS/eip-6353.md

```diff
@@ -0,0 +1,333 @@
+
+    address[] whitelistedAddresses; //Addresses whitelisted
+
+    event AddedToWhitelist (address toAdd);
+    event RemovedFromWhitelist (address toRemove);
+    event DonnationAddressChanged (address whitelistedAddr);
+    event DonnationAddressAndRateChanged (address whitelistedAddr,uint256 rate);
+    event ModifiedCharityRate(address whitelistedAddr,uint256 rate)
+
+```
+
+### Functions
+
+#### **addToWhitelist**
+
+Add address to whitelist and set the rate to the default rate.
+
+| Parameter | Description |
+| ---------|-------------|
+| toAdd | The address to the whitelist.
+
+#### **deleteFromWhitelist**
+
+Remove the address from the whitelist and set rate to the default rate.
+
+| Parameter | Description |
+| ---------|-------------|
+| toRemove | The address to remove from whitelist.
+
+#### **getAllWhitelistedAddresses**
+
+Get all registered charity addresses.
+
+#### **getRate**
+
+Display for a user the rate of the default charity address that will receive donation.
+
+#### **setSpecificRate**
+
+Set personalized rate for charity address in {whitelistedRate}.
+
+| Parameter | Description |
+| ---------|-------------|
+| whitelistedAddr | The address to set as default. |
+| rate  | The personalised rate for donation. |
+
+#### **_returnRate**
+
+Return the rate to donate.
+
+| Parameter | Description |
+| ---------|-------------|
+| from | The address to get the rate of donation.
+
+#### **setSpecificDefaultAddress**
+
+Set for a user a default charity address that will receive donations. The default rate specified in {whitelistedRate} will be applied.
+
+| Parameter | Description |
+| ---------|-------------|
+| whitelistedAddr | The address to set as default.
+
+#### **setSpecificDefaultAddressAndRate**
+
+Set for a user a default charity address that will receive donations. The rate is specified by the user.
+
+| Parameter | Description |
+| ---------|-------------|
+| whitelistedAddr | The address to set as default. |
+| rate  | The personalized rate for donation.
+
+#### **specificDefaultAddress**
+
+Display for a user the default charity address that will receive donations. The default rate specified in {whitelistedRate} will be applied.
+
+#### **deleteDefaultAddress**
+
+Delete The Default Address and so deactivate donations.
+
+#### **charityInfo**
+
+Called with the charity address to determine if the contract whitelisted the address and if it is, the rate assigned.
+
+| Parameter | Description |
+| ---------|-------------|
+| addr | The Charity address queried for donnation information.
+
+## Rationale
+
+This standard provides functionality that allows token holders to donate easily. The donation when activated is done directly in the overridden `transfer`, `transferFrom`, and `approve` functions.
+
+The donation is a percentage-based rate model, but the calculation can be done differently. Also, donations could be in other functions than the `transfer` function.
```

Comment:
This line reads more like specification than Rationale. You're listing things that are permitted, not explaining choices you've made.